pkg/host/config.go

```
@@ -94,112 +84,20 @@ type SonrHost interface {
-	Status_STANDBY                      // Host is standby, waiting for connection
-	Status_CONNECTING                   // Host is connecting
-	Status_READY                        // Host is ready
-	Status_FAIL                         // Host failed to connect
-	Status_CLOSED                       // Host is closed
-)
-
-// Equals returns true if given SNRHostStatus matches this one
-func (s HostStatus) Equals(other HostStatus) bool {
-	return s == other
-}
-
-// IsNotIdle returns true if the SNRHostStatus != Status_IDLE
-func (s HostStatus) IsNotIdle() bool {
-	return s != Status_IDLE
-}
-
-// IsStandby returns true if the SNRHostStatus == Status_STANDBY
-func (s HostStatus) IsStandby() bool {
-	return s == Status_STANDBY
-}
-
-// IsReady returns true if the SNRHostStatus == Status_READY
-func (s HostStatus) IsReady() bool {
-	return s == Status_READY
-}
-
-// IsConnecting returns true if the SNRHostStatus == Status_CONNECTING
-func (s HostStatus) IsConnecting() bool {
-	return s == Status_CONNECTING
-}
-
-// IsFail returns true if the SNRHostStatus == Status_FAIL
-func (s HostStatus) IsFail() bool {
-	return s == Status_FAIL
-}
-
-// IsClosed returns true if the SNRHostStatus == Status_CLOSED
-func (s HostStatus) IsClosed() bool {
-	return s == Status_CLOSED
-}
-
-// String returns the string representation of the SNRHostStatus
-func (s HostStatus) String() string {
-	switch s {
-	case Status_IDLE:
-		return "IDLE"
-	case Status_STANDBY:
-		return "STANDBY"
-	case Status_CONNECTING:
-		return "CONNECTING"
-	case Status_READY:
-		return "READY"
-	case Status_FAIL:
-		return "FAIL"
-	case Status_CLOSED:
-		return "CLOSED"
-	}
-	return "UNKNOWN"
-}
+	VerifyData(data []byte, signature []byte, peerId peer.ID, pubKeyData []byte) error
 
-// SetStatus sets the host status and emits the event
-func (h *hostImpl) SetStatus(s HostStatus) {
-	// Check if status is changed
-	if h.status == s {
-		return
-	}
+	// Close closes the node
+	Close()
```

I think most of these should return `error`, because their current implementations just suppress any errors that pop up. Might make debugging harder.

---

thats fair, i didn't refactor the interfaces in this implementation. perhaps that is a separate task?